proj/src/main/webapp/chat.js

```diff
@@ -924,6 +924,35 @@ function addUserInfoToDom(userObj) {
     }
 }
 
+function initDeleteAccountButton(){
+    const txtEmail = document.getElementById("email");
+    const txtPassword = document.getElementById("pass");
+    const btnDelete = document.getElementById("btnDeleteAccount");
+    
+    
+
+    btnDelete.addEventListener("click", async function () {
+        const emailVal = txtEmail.value;
+        const passVal = txtPassword.value;
+
+        console.log(emailVal);
+
+        // Re-authenticate credentials
+        const auth = firebase.auth();
+        const promise = auth.signInWithEmailAndPassword(emailVal, passVal).then(function(){
+            // Deletes user from database but not their messages 
+            // A new account can be created with the previously associated email 
+            var user = firebase.auth().currentUser;
+            user.delete().then(function() {
+                window.location.replace(welcome.html);
```

Comment:
Also, here right before you call user.delete(), can you sign the user out so that they no longer receive push notifications? I think there's a function for that at the bottom of chat.js

Reply:
You might be able to just call `notifications.unSubscribeFromAllChats()`

Reply:
just updated it